internal/balancer/balancer.go

```diff
@@ -71,7 +65,12 @@ func (b *Balancer) clusterDiscovery(ctx context.Context) (err error) {
 		)
 	}()
 
-	endpoints, err = b.discovery.Discover(ctx)
+	client := b.discoveryClient()
+	if err != nil {
```

Comment:
what error are we checking in this place? (where did you get it from?)

Reply:
ack